app/components/password-reset/index.ts

```diff
@@ -0,0 +1,122 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import ENV from 'irene/config/environment';
+import { task } from 'ember-concurrency';
+
+import lookupValidator from 'ember-changeset-validations';
+import { Changeset } from 'ember-changeset';
+import { BufferedChangeset } from 'ember-changeset/types';
+
+import {
+  validatePresence,
+  validateConfirmation,
+} from 'ember-changeset-validations/validators';
+
+import IntlService from 'ember-intl/services/intl';
+import RouterService from '@ember/routing/router-service';
+
+interface PasswordResetSignature {
+  Args: {
+    token?: string;
+  };
+}
+
+type ChangesetBufferProps = Partial<BufferedChangeset> & {
+  password: string;
+  confirm_password: string;
+};
+
+const ResetValidator = {
```

Comment:
This should be moved to the validations folder.

Reply:
hmm I think we should move away from this and keep validators close to the components unless it is going to be reused. if the `validator` is small then keep it in same `ts` file otherwise create a file `validator.ts` in the same component folder